### ISSUE TITLE
Custom Variable Labels

### ISSUE THREAD

**Issue by Rae309** [kassambara/factoextra#71]
I'm using fviz_pca_biplot, but I would like to use custom labels for the variable names that go over the arrows so that I can include subscript, superscript etc.  Is there a way to do this?

**Comment by duttashi** [comment]
@Rae309 from the function signature, `help("fviz_pca_biplot")`, I do not think adding custom labels is possible. The most you can do is to include `geom = c("text")` that will add labels over the arrows. A minimum reprex is given below; @kassambara maybe this can be a future enhancement?

```
data(iris)
res.pca <- prcomp(iris[, -5],  scale = TRUE)
fviz_pca_biplot(res.pca, label = "var", 
                habillage=iris$Species,
                addEllipses=TRUE, ellipse.level=0.95,
                geom = c("text"),
                ggtheme = theme_minimal())
```

**Comment by AldridgeCaleb** [comment]
@Rae309 if you are using the `princomp` or `prcomp` functions from the `stats` package, then you can change variables names by updating the row names in `$loadings` (`princomp`) or `$rotation` (`prcomp`). Extending the reprex provided by @duttashi, the variable names are changed from "Sepal.Length", "Sepal.Width", "Petal.Length", and "Petal.Width" to "Sepal Length (mm)", "SEP W", "PL", and "petal_width".
```
data(iris)  
res.pca <- prcomp(iris[, -5],  scale = TRUE)  
row.names(res.pca$rotation) <-  c("Sepal Length (mm)", "SEP W", "PL", "petal_width")  
fviz_pca_biplot(res.pca, label = "var",  
  habillage = iris$Species,  
  addEllipses = TRUE, ellipse.level = 0.95,  
  geom = c("text"),  
  ggtheme = theme_minimal()  
)
```

**Comment by ArkosicGreywacke** [comment]
> @Rae309 if you are using the `princomp` or `prcomp` functions from the `stats` package, then you can change variables names by updating the row names in `$loadings` (`princomp`) or `$rotation` (`prcomp`). Extending the reprex provided by @duttashi, the variable names are changed from "Sepal.Length", "Sepal.Width", "Petal.Length", and "Petal.Width" to "Sepal Length (mm)", "SEP W", "PL", and "petal_width".
> 
> ```
> data(iris)  
> res.pca <- prcomp(iris[, -5],  scale = TRUE)  
> row.names(res.pca$rotation) <-  c("Sepal Length (mm)", "SEP W", "PL", "petal_width")  
> fviz_pca_biplot(res.pca, label = "var",  
>   habillage = iris$Species,  
>   addEllipses = TRUE, ellipse.level = 0.95,  
>   geom = c("text"),  
>   ggtheme = theme_minimal()  
> )
> ```

Any suggestions for special characters like greek letters, superscripts and subscripts?

**Comment by GavinDuley** [comment]
Is there any update on this? It would be particularly useful where variable names include characters that R objects to when included in column names.

**Comment by kassambara** [comment]
Yes — the variable labels are taken from the **rownames of the loadings**, so you can set any custom labels by renaming them before plotting (`$rotation` for `prcomp`, `$loadings` for `princomp`):

``` r
suppressWarnings(suppressPackageStartupMessages(library(factoextra)))
res.pca <- prcomp(iris[, -5], scale. = TRUE)
# Variable labels come from the loadings rownames -> rename them for custom labels:
rownames(res.pca$rotation) <- c("Sepal length (mm)", "Sepal width (mm)",
                                "Petal length (mm)", "Petal width (mm)")
fviz_pca_var(res.pca, repel = TRUE, title = "Custom variable labels via rownames()")
```

![](https://i.imgur.com/lqs2EIV.png)

<sup>Created on 2026-06-25 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

For subscripts/superscripts/Greek you can additionally style the text layer with the **ggtext** package (`ggplot2::theme(... = ggtext::element_markdown())`) or post-process the label layer with `parse = TRUE` for plotmath. Closing — reopen if you need a worked plotmath example.
